lib/ansible/runner/lookup_plugins/dig.py

```diff
@@ -0,0 +1,124 @@
+            example.com/TXT                                 # specific qtype
+            192.168.1.2/PTR                                 # reverse PTR
+              ^^ shortcut for 2.1.168.192.in-addr.arpa/PTR
+            example.net/AAAA  @nameserver                   # query specified server
+                               ^^^ can be comma-sep list of names/addresses
+        '''
+        terms = terms.split()
+
+        # Create Resolver object so that we can set NS if necessary
+        myres = dns.resolver.Resolver()
+
+        domain = None
+        qtype  = 'A'
+
+        for t in terms:
+            if t.startswith('@'):       # e.g. "@10.0.1.2,192.168.1.1" is ok.
+                nsset = t[1:].split(',')
+                nameservers = []
+                for ns in nsset:
+                    # Check if we have a valid IP address. If so, use that, otherwise
+                    # try to resolve name to address using system's resolver. If that
+                    # fails we bail out.
+                    try:
+                        socket.inet_aton(ns)
+                        nameservers.append(ns)
+                    except:
+                        try:
+                            nsaddr = dns.resolver.query(ns)[0].address
+                            nameservers.append(nsaddr)
+                        except Exception, e:
+                            raise errors.AnsibleError("dns lookup NS: ", str(e))
+                    myres.nameservers = nameservers
+                continue
+            if '/' in t:
+                try:
+                    domain, qtype = t.split('/')
+                except:
+                    domain = t
+            else:
+                domain = t
+
+        # print "--- domain = {0} qtype={1}".format(domain, qtype)
+
+        ret = []
+
+        if qtype.upper() == 'PTR':
+            try:
+                n = dns.reversename.from_address(domain)
+                domain = n.to_text()
+            except dns.exception.SyntaxError:
+                pass
+            except Exception, e:
+                pass
+                print "START"
+                # raise errors.AnsibleError("dns.reversename unhandled exception", str(e))
+
+        responses = []
+        try:
+            answers = myres.query(domain, qtype)
+            for rdata in answers:
+                s = rdata.to_text()
+                if qtype.upper() == 'TXT':
+                    s = s[1:-1]  # Strip outside quotes on TXT rdata
+                responses.append(s)
+
+        except dns.resolver.NXDOMAIN:
+            responses.append('NXDOMAIN')
+        except dns.resolver.NoAnswer:
+            responses.append("")
+        except dns.resolver.Timeout:
+            responses.append('')
+        except dns.exception.DNSException, e:
+            raise errors.AnsibleError("dns.resolver unhandled exception", e)
+
+        for r in responses:
+            ret.append(r)
+        return ret
```

Comment:
I'm not sure ret/responses are in the correct scope, it looks like you wanted responses inside a for loop (terms one?) in which case indentation is wrong.